Core/src/RequestTrait.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Core;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Utils;
+
+/**
+ * Provides a light wrapper around often used request modification functions.
+ *
+ * @internal
+ */
+trait RequestTrait
```

Comment:
Generally speaking, we should only add traits if it contains code we use in more than one place, which is not the case here.

Reply:
https://github.com/googleapis/google-cloud-php/pull/6055 ?